app/javascript/src/NewService/components/FormTemplate.jsx

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+import React from 'react'
+import {CSRFToken} from 'utilities/utils'
+import type {FormTemplateProps} from 'NewService/types'
+
+type ServiceDiscoveryFormTemplateProps = {
+  hiddenServiceDiscoveryInput: {
+    value: string,
+    name: string,
+    id: string
+  },
+  serviceDiscoverySelect: {
+    [string]: {
+        htmlFor: string,
+        label: string,
+        name: string,
+        id: string
+    }
+  }
+}
+
+const serviceDiscoveryProps: ServiceDiscoveryFormTemplateProps = {
+  hiddenServiceDiscoveryInput: {
+    value: 'discover',
+    name: 'service[source]',
+    id: 'service_source'
+  },
+  serviceDiscoverySelect: {
+    project: {
+      htmlFor: 'namespace',
+      label: 'Namespace',
+      name: 'service[namespace]',
+      id: 'service_namespace'
+    },
+    service: {
+      htmlFor: 'service_name',
+      label: 'Name',
+      name: 'service[name]',
+      id: 'service_name'
+    }
+  }
+}
+
+const renderLabel = (project: {htmlFor: string, label: string}, required: boolean = false) => <label htmlFor={project.htmlFor}>{project.label}
```

Comment:
Minor: the body of the fn could start in the next line, it will be easier to read. This applies to the rest.